packages/synapse-react-client/src/components/SynapseHomepageV2/SynapseInAction.tsx

```diff
@@ -0,0 +1,164 @@
+    selectColumn => selectColumn.name == 'title',
+  )!
+  const descriptionColIndex = headers?.findIndex(
+    selectColumn => selectColumn.name == 'description',
+  )!
+  const tagsColIndex = headers?.findIndex(
+    selectColumn => selectColumn.name == 'tags',
+  )!
+  const imageFileColIndex = headers?.findIndex(
+    selectColumn => selectColumn.name == 'image',
+  )!
+  const mobileImageFileColIndex = headers?.findIndex(
+    selectColumn => selectColumn.name == 'mobileImage',
+  )!
+  const logoColIndex = headers?.findIndex(
+    selectColumn => selectColumn.name == 'logo',
+  )!
+  const linkColIndex = headers?.findIndex(
+    selectColumn => selectColumn.name == 'url',
+  )!
+  const friendlyNameColIndex = headers?.findIndex(
+    selectColumn => selectColumn.name == 'friendlyName',
+  )!
+  const primaryColorColIndex = headers?.findIndex(
+    selectColumn => selectColumn.name == 'primaryColor',
+  )!
+  const secondaryColorColIndex = headers?.findIndex(
+    selectColumn => selectColumn.name == 'secondaryColor',
+  )!
+
+  if (!rowSet || rowSet.rows.length == 0) {
+    return <></>
+  }
+  return (
+    <Box sx={{ position: 'relative' }} ref={ref}>
+      <Box sx={{ width: { xs: '100%', sm: '45%' } }}>
+        <Box
+          sx={{
+            maxWidth: '450px',
+            m: 'auto',
+            zIndex: 100,
+            position: 'relative',
+            backgroundColor: 'rgba(245, 249, 249, .8)',
+            borderRadius: '12px',
+          }}
+        >
+          {rowSet.rows.map((row, index) => {
+            const title = row.values[titleColIndex]!
+            const description = row.values[descriptionColIndex]!
+            const tags: string[] = JSON.parse(row.values[tagsColIndex]!)
+            const imageFileHandleId = row.values[imageFileColIndex]!
+            const mobileImageFileHandleId = row.values[mobileImageFileColIndex]!
+            const logoFileHandleId = row.values[logoColIndex]!
+            const link = row.values[linkColIndex]!
+            const friendlyName = row.values[friendlyNameColIndex]!
+            const primaryColor = row.values[primaryColorColIndex]!
+            const secondaryColor = row.values[secondaryColorColIndex]!
+            const backgroundColor =
+              isMobileView && index % 2 ? '#DAE9E7' : undefined
+            return (
+              <Box
+                key={row.rowId}
+                sx={{
+                  pt: { xs: '30px', sm: '120px' },
+                  pb: { xs: '30px', sm: '120px' },
+                  backgroundColor: { backgroundColor },
+                }}
+              >
+                {/* Preload the desktop image file handle ID so it is ready when the user scrolls down to view it */}
+                <Box
+                  sx={{
+                    display: 'none',
+                  }}
+                >
+                  <ImageFromSynapseTable
+                    tableId={tableId}
+                    fileHandleId={imageFileHandleId}
```

Comment:
I wonder if there's a more elegant way to preload (maybe using `rootMargin` arg on `useInView`?) without rendering a hidden object, but this whole set of interactions in this component is fairly complex so you may already have discovered one of the best ways to do it.